.github/workflows/check-catalyst.yaml

```diff
@@ -139,57 +147,57 @@ jobs:
 
   mhlo:
     name: MHLO Dialect Build
-    needs: [constants, llvm]
-    runs-on: ubuntu-latest
+    needs: [constants, llvm, determine_runner]
+    runs-on: ${{ needs.determine_runner.outputs.runner_group }}
     strategy:
       matrix:
         compiler: ${{ fromJson(needs.constants.outputs.compilers) }}
 
     steps:
     - name: Checkout Catalyst repo
-      uses: actions/checkout@v3
+      uses: actions/checkout@v4
 
     - name: Cache MHLO Source
       id: cache-mhlo-source
-      uses: actions/cache@v3
+      uses: actions/cache@v4
       with:
         path: mlir/mlir-hlo
         key: ${{ runner.os }}-mhlo-${{ needs.constants.outputs.mhlo_version }}-default-source
-        enableCrossOsArchive: True
+        enableCrossOsArchive: true
 
     - name: Clone MHLO Submodule
       if: steps.cache-mhlo-source.outputs.cache-hit != 'true'
-      uses: actions/checkout@v3
+      uses: actions/checkout@v4
       with:
         repository: tensorflow/mlir-hlo
         ref: ${{ needs.constants.outputs.mhlo_version }}
         path: mlir/mlir-hlo
 
     - name: Cache MHLO Build
       id: cache-mhlo
-      uses: actions/cache@v3
+      uses: actions/cache@v4
       with:
         path: mhlo-build
         key: ${{ runner.os }}-mhlo-${{ needs.constants.outputs.mhlo_version }}-default-build-${{ matrix.compiler }}
 
     - name: Get Cached LLVM Source
       id: cache-llvm-source
       if: steps.cache-mhlo.outputs.cache-hit != 'true'
-      uses: actions/cache@v3
+      uses: actions/cache@v4
       with:
         path: mlir/llvm-project
         key: ${{ runner.os }}-llvm-${{ needs.constants.outputs.llvm_version }}-default-source
-        enableCrossOsArchive: True
-        fail-on-cache-miss: True
+        enableCrossOsArchive: true
+        fail-on-cache-miss: true
```

Comment:
Is `True` is not the correct keyword for toml? :sweat_smile:

Reply:
This is `yaml`, but also here is the github docs for it: https://docs.github.com/en/actions/learn-github-actions/expressions#literals